[Bug] win客户端缓存问题导致jellyfin一直命中错误的缓存，无法正常连接

### 提交前确认

- [x] 我已经搜索过现有 Issues，确认这不是重复问题
- [x] 我已经阅读过故障排查文档，并尝试过其中适用的方案
- [ ] 如果条件允许，我已经在较新版本上再次确认过这个问题

### 平台

Windows

### 系统版本

windows11 24H2

### NipaPlay 版本

1.9.22

### 安装来源

GitHub Releases

### 相关功能模块

Jellyfin

### 播放内核（如适用）

FVP (libmdk)

### 问题描述

**此配置中所有的jellyfin都是同一台jellyfin服务器，只是连接方式不同。此问题出现在win上，linux上没有复现**

jellyfin 第一次配置出错了之后，配置会缓存在`%APPDATA%\Roaming\com.aimessoft\nipaplay` 中。
后续修改jellyfin的配置，导致一直命中错误的缓存配置。

结果：
** 详细的查看日志部分 **
`http://[fc03:1136:3831:2855:5cc2:8e8:9d29:0]:8096` 是我第二次需要添加的jellyfin地址，但是后续app会去找缓存配置，找到了 `http://jellyfinsingle.xiaozi.heiyu.space` 这个缓存，并成功命中（这个是第一次添加jellyfin地址的时候配置错误的）。这个流程应该是有问题的（命中缓存没有考虑到之后的配置是否是有效的，直接命中导致后续无法修改配置）

问题：
添加同一台jellyfin服务器的配置缓存命中是怎样的？
如果之前的配置失效了，怎么确保后续添加同一台jellyfin的服务器不会命中错误的配置？

### 复现步骤

windows 版本 1.9.22 添加 jellyfin 配置

1. 第一次添加jellyfin配置时添加一个不可用的 (如：应该是https的，却配置成了http了)
2. 第一次添加配置失败之后进行第二次添加配置（使用正确的配置）

### 预期行为

第二次添加配置的时候应该使用第二次的配置去连接jellyfin服务，而不是通过命中的`错误的缓存`去连接

### 日志或文本报错

```shell
txt
2026-04-15 17:30:09.163 [DEBUG] [App] 检查更新完成: 当前版本=1.9.22, 最新版本=1.9.22, 有更新=false
2026-04-15 17:30:09.163 [DEBUG] [App] 检查更新时出现错误: 无法获取版本信息
2026-04-15 17:30:26.621 [INFO] [App] MultiAddressServer: 加载了 1 个服务器配置
2026-04-15 17:30:26.621 [INFO] [Network] Jellyfin: 开始连接服务器 http://[fc03:1136:3831:2855:5cc2:8e8:9d29:0]:8096 (用户: root, 地址名: 公网)
2026-04-15 17:30:26.621 [INFO] [App] MultiAddressServer: 开始识别服务器 - URL: http://[fc03:1136:3831:2855:5cc2:8e8:9d29:0]:8096, Type: jellyfin
2026-04-15 17:30:26.638 [INFO] [App] MultiAddressServer: 获取到服务器ID - 50424104be804f609187875fe2e607cd
2026-04-15 17:30:26.638 [INFO] [App] MultiAddressServer: 搜索现有配置，总数: 1
2026-04-15 17:30:26.638 [INFO] [App] MultiAddressServer: 检查配置 - Type: jellyfin, ServerId: 50424104be804f609187875fe2e607cd, Username: root
2026-04-15 17:30:26.638 [INFO] [App] MultiAddressServer: 通过serverId找到现有配置: 1776234653267
2026-04-15 17:30:26.638 [INFO] [App] MultiAddressServer: 识别成功 - ServerId: 50424104be804f609187875fe2e607cd, 现有配置: 1776234653267
2026-04-15 17:30:26.638 [INFO] [App] MultiAddressServer: 尝试连接 公网 (http://jellyfinsingle.xiaozi.heiyu.space)
2026-04-15 17:30:26.662 [INFO] [App] MultiAddressServer: 保存了 1 个服务器配置
2026-04-15 17:30:26.662 [INFO] [App] MultiAddressServer: 成功连接到 公网 (http://jellyfinsingle.xiaozi.heiyu.space)
2026-04-15 17:30:26.671 [ERROR] [Network] Jellyfin: 连接失败: Exception: 认证失败: 307
```

### 补充信息

jellyfin10.x
windows nipaplay 1.9.22

--- a/lib/services/dandanplay_service_io.dart
+++ b/lib/services/dandanplay_service_io.dart
@@ -141,9 +141,14 @@ class DandanplayService {
     return;
   }
 
-  /// 获取当前弹弹play API 基础 URL（包含用户自定义设置）
+  /// 获取弹幕相关 API 基础 URL（包含用户自定义设置）
   static Future<String> getApiBaseUrl() async {
     return await NetworkSettings.getDandanplayServer();
+  }
+
+  /// 获取账号相关 API 基础 URL（固定官方服务器）
+  static Future<String> getAccountApiBaseUrl() async {
+    return NetworkSettings.primaryServer;
   }
 
   // 预加载最近更新的动画数据
@@ -249,7 +254,7 @@ class DandanplayService {
         final timestamp = (DateTime.now().toUtc().millisecondsSinceEpoch / 1000)
             .round();
 
-        final apiBaseUrl = await getApiBaseUrl();
+        final apiBaseUrl = await getAccountApiBaseUrl();
         final requestUri = Uri.parse('$apiBaseUrl$apiPath');
         final headers = _buildLoginRenewHeaders(
           timestamp: timestamp,
@@ -489,7 +494,7 @@ class DandanplayService {
       final hash = md5.convert(utf8.encode(hashString)).toString();
 
       final response = await http.post(
-        Uri.parse('${await getApiBaseUrl()}/api/v2/login'),
+        Uri.parse('${await getAccountApiBaseUrl()}/api/v2/login'),
         headers: {
           'Content-Type': 'application/json',
           'Accept': 'application/json',
@@ -564,7 +569,7 @@ class DandanplayService {
       final timestamp = (DateTime.now().toUtc().millisecondsSinceEpoch / 1000)
           .round();
       const apiPath = '/api/v2/oauthprovider/bangumi/login';
-      final baseUrl = await getApiBaseUrl();
+      final baseUrl = await getAccountApiBaseUrl();
       final normalizedRedirect = redirectUrl?.trim();
       final query = <String, String>{};
       if (normalizedRedirect != null && normalizedRedirect.isNotEmpty) {
@@ -626,7 +631,7 @@ class DandanplayService {
     }
 
     try {
-      final apiBaseUrl = await getApiBaseUrl();
+      final apiBaseUrl = await getAccountApiBaseUrl();
       const apiPath = '/api/v2/login/renew';
       final appSecret = await getAppSecret();
       final timestamp = (DateTime.now().toUtc().millisecondsSinceEpoch / 1000)
@@ -804,7 +809,7 @@ class DandanplayService {
         appSecret,
       );
       final response = await http.post(
-        Uri.parse('${await getApiBaseUrl()}/api/v2/register'),
+        Uri.parse('${await getAccountApiBaseUrl()}/api/v2/register'),
         headers: {
           'Content-Type': 'application/json',
           'Accept': 'application/json',
@@ -869,7 +874,7 @@ class DandanplayService {
       const apiPath = '/api/v2/playhistory';
 
       final response = await http.post(
-        Uri.parse('${await getApiBaseUrl()}$apiPath'),
+        Uri.parse('${await getAccountApiBaseUrl()}$apiPath'),
         headers: {
           'Content-Type': 'application/json',
           'Accept': 'application/json',
@@ -1662,7 +1667,7 @@ class DandanplayService {
         queryParams['toDate'] = toDate.toUtc().toIso8601String();
       }
 
-      final baseUrl = await getApiBaseUrl();
+      final baseUrl = await getAccountApiBaseUrl();
       final uri = Uri.parse(
         '$baseUrl$apiPath${queryParams.isNotEmpty ? '?' + Uri(queryParameters: queryParams).query : ''}',
       );
@@ -1743,7 +1748,7 @@ class DandanplayService {
       debugPrint('[弹弹play服务] 提交播放历史: $episodeIdList');
 
       final response = await http.post(
-        Uri.parse('${await getApiBaseUrl()}$apiPath'),
+        Uri.parse('${await getAccountApiBaseUrl()}$apiPath'),
         headers: {
           'Content-Type': 'application/json',
           'Accept': 'application/json',
@@ -1903,7 +1908,7 @@ class DandanplayService {
         queryParams['onlyOnAir'] = 'true';
       }
 
-      final baseUrl = await getApiBaseUrl();
+      final baseUrl = await getAccountApiBaseUrl();
       final uri = Uri.parse(
         '$baseUrl$apiPath${queryParams.isNotEmpty ? '?' + Uri(queryParameters: queryParams).query : ''}',
       );
@@ -1978,7 +1983,7 @@ class DandanplayService {
       debugPrint('[弹弹play服务] 添加收藏: animeId=$animeId, status=$favoriteStatus');
 
       final response = await http.post(
-        Uri.parse('${await getApiBaseUrl()}$apiPath'),
+        Uri.parse('${await getAccountApiBaseUrl()}$apiPath'),
         headers: {
           'Content-Type': 'application/json',
           'Accept': 'application/json',
@@ -2031,7 +2036,7 @@ class DandanplayService {
       debugPrint('[弹弹play服务] 取消收藏: animeId=$animeId');
 
       final response = await http.delete(
-        Uri.parse('${await getApiBaseUrl()}$apiPath'),
+        Uri.parse('${await getAccountApiBaseUrl()}$apiPath'),
         headers: {
           'Accept': 'application/json',
           'User-Agent': userAgent,
@@ -2245,7 +2250,7 @@ class DandanplayService {
       final apiPath = '/api/v2/oauth/webToken';
 
       final response = await http.get(
-        Uri.parse('${await getApiBaseUrl()}$apiPath?business=$business'),
+        Uri.parse('${await getAccountApiBaseUrl()}$apiPath?business=$business'),
         headers: {
           'Accept': 'application/json',
           'User-Agent': userAgent,
@@ -2366,7 +2371,7 @@ class DandanplayService {
 
       // 2. 构建注销页面URL
       final deleteAccountUrl =
-          '${await getApiBaseUrl()}/api/v2/oauth/deleteAccount?webToken=$webToken';
+          '${await getAccountApiBaseUrl()}/api/v2/oauth/deleteAccount?webToken=$webToken';
 
       debugPrint('[弹弹play服务] 账号注销URL: $deleteAccountUrl');
 
@@ -2397,7 +2402,7 @@ class DandanplayService {
       }
 
       final manageUri = Uri.parse(
-              '${await getApiBaseUrl()}/api/v2/oauthprovider/bangumi/manage')
+              '${await getAccountApiBaseUrl()}/api/v2/oauthprovider/bangumi/manage')
           .replace(queryParameters: {'webToken': webToken});
       final manageUrl = manageUri.toString();
       debugPrint('[弹弹play服务] Bangumi管理页URL: $manageUrl');

--- a/lib/services/media_server_service_base.dart
+++ b/lib/services/media_server_service_base.dart
@@ -329,6 +329,10 @@ abstract class MediaServerServiceBase {
         } else {
           print('$serviceNameService: 地址已存在，使用现有配置');
         }
+
+        if (profile != null && profile.username != username) {
+          profile = profile.copyWith(username: username);
+        }
       } else if (identifyResult.isConflict) {
         print('$serviceNameService: 检测到冲突，抛出异常: ${identifyResult.error}');
         logService.addLog(
@@ -365,6 +369,7 @@ abstract class MediaServerServiceBase {
       final connectionResult = await _multiAddressService.tryConnect(
         profile: profile,
         testConnection: (url) => testConnection(url, username, password),
+        preferredUrl: normalizedUrl,
       );
 
       if (connectionResult.success && connectionResult.profile != null) {

--- a/lib/services/multi_address_server_service.dart
+++ b/lib/services/multi_address_server_service.dart
@@ -280,10 +280,25 @@ class MultiAddressServerService {
   Future<ConnectionResult> tryConnect({
     required ServerProfile profile,
     required Future<bool> Function(String url) testConnection,
+    String? preferredUrl,
   }) async {
     final addresses = profile.enabledAddresses;
+    final normalizedPreferredUrl = preferredUrl == null
+        ? null
+        : _normalizeUrl(preferredUrl);
+
+    final orderedAddresses = normalizedPreferredUrl == null
+        ? addresses
+        : [
+            ...addresses.where(
+              (address) => address.normalizedUrl == normalizedPreferredUrl,
+            ),
+            ...addresses.where(
+              (address) => address.normalizedUrl != normalizedPreferredUrl,
+            ),
+          ];
     
-    if (addresses.isEmpty) {
+    if (orderedAddresses.isEmpty) {
       return ConnectionResult(
         success: false,
         error: '没有可用的服务器地址',
@@ -294,7 +309,7 @@ class MultiAddressServerService {
     ServerProfile updatedProfile = profile;
     
     // 按优先级尝试每个地址
-    for (final address in addresses) {
+    for (final address in orderedAddresses) {
       if (!address.shouldRetry()) {
         DebugLogService().addLog(
           'MultiAddressServer: 跳过地址 ${address.name} (${address.url}) - 失败次数过多'

--- a/lib/utils/danmaku_dialog_manager.dart
+++ b/lib/utils/danmaku_dialog_manager.dart
@@ -119,6 +119,13 @@ class DanmakuDialogManager {
   // 注册发送弹幕快捷键处理
   bool handleSendDanmakuHotkey() {
     debugPrint('[DanmakuDialogManager] 处理发送弹幕快捷键，当前对话框状态: ${_isShowingDialog ? "显示中" : "未显示"}');
+
+    final focusedContext = FocusManager.instance.primaryFocus?.context;
+    final isEditingText = focusedContext?.widget is EditableText;
+    if (_isShowingDialog && isEditingText) {
+      debugPrint('[DanmakuDialogManager] 检测到输入焦点，忽略关闭弹幕对话框快捷键');
+      return true;
+    }
     
     if (_isShowingDialog) {
       // 如果已经在显示弹幕对话框，则关闭它

--- a/lib/utils/hotkey_service.dart
+++ b/lib/utils/hotkey_service.dart
@@ -228,6 +228,9 @@ class HotkeyService extends ChangeNotifier {
       await hotKeyManager.register(
         hotKey,
         keyDownHandler: (HotKey hotKey) {
+          if (_shouldBlockHotkeyInTextInput()) {
+            return;
+          }
           ////debugPrint('[HotkeyService] 热键触发: $description ($keyString)');
           handler();
         },
@@ -238,6 +241,43 @@ class HotkeyService extends ChangeNotifier {
     } catch (e) {
       ////debugPrint('[HotkeyService] 注册热键失败 $description ($keyString): $e');
     }
+  }
+
+  bool _isEditableTextFocused() {
+    final focusedContext = FocusManager.instance.primaryFocus?.context;
+    if (focusedContext == null) {
+      return false;
+    }
+    return focusedContext.widget is EditableText;
+  }
+
+  bool _shouldBlockHotkeyInTextInput() {
+    if (_isEditableTextFocused()) {
+      return true;
+    }
+    return false;
+  }
+
+  bool _isShiftPressed() {
+    final pressed = HardwareKeyboard.instance.logicalKeysPressed;
+    return pressed.contains(LogicalKeyboardKey.shiftLeft) ||
+        pressed.contains(LogicalKeyboardKey.shiftRight) ||
+        pressed.contains(LogicalKeyboardKey.shift);
+  }
+
+  bool _isShiftShortcutConfigured(String action, PhysicalKeyboardKey arrowKey) {
+    final shortcut = _shortcuts[action];
+    if (shortcut == null || !shortcut.contains('Shift+')) {
+      return false;
+    }
+    final info = _parseKeyString(shortcut);
+    if (info == null) {
+      return false;
+    }
+    if (!info.modifiers.contains(HotKeyModifier.shift)) {
+      return false;
+    }
+    return info.keyCode == arrowKey;
   }
 
   // 特别处理ESC键
@@ -598,6 +638,14 @@ class HotkeyService extends ChangeNotifier {
   }
 
   void _handleRewind() {
+    if (_isShiftPressed() &&
+        _isShiftShortcutConfigured(
+          'previous_episode',
+          PhysicalKeyboardKey.arrowLeft,
+        )) {
+      _handlePreviousEpisode();
+      return;
+    }
     final videoState = _getVideoPlayerState();
     if (videoState != null) {
       videoState.seekBackwardByStep();
@@ -605,6 +653,14 @@ class HotkeyService extends ChangeNotifier {
   }
 
   void _handleForward() {
+    if (_isShiftPressed() &&
+        _isShiftShortcutConfigured(
+          'next_episode',
+          PhysicalKeyboardKey.arrowRight,
+        )) {
+      _handleNextEpisode();
+      return;
+    }
     final videoState = _getVideoPlayerState();
     if (videoState != null) {
       videoState.seekForwardByStep();
@@ -705,6 +761,9 @@ class HotkeyService extends ChangeNotifier {
   }
 
   void _handleForwardKeyDown() {
+    if (_shouldBlockHotkeyInTextInput()) {
+      return;
+    }
     _isForwardKeyPressed = true;
 
     // 取消之前的计时器
@@ -719,6 +778,9 @@ class HotkeyService extends ChangeNotifier {
   }
 
   void _handleForwardKeyUp() {
+    if (_shouldBlockHotkeyInTextInput()) {
+      return;
+    }
     _isForwardKeyPressed = false;
 
     // 取消长按计时器
@@ -752,6 +814,9 @@ class HotkeyService extends ChangeNotifier {
   }
 
   void _handleEscape() {
+    if (_shouldBlockHotkeyInTextInput()) {
+      return;
+    }
     final videoState = _getVideoPlayerState();
     if (videoState != null && videoState.isFullscreen) {
       videoState.toggleFullscreen();

--- a/lib/utils/hotkey_service.dart
+++ b/lib/utils/hotkey_service.dart
@@ -274,7 +274,9 @@ class HotkeyService extends ChangeNotifier {
     if (info == null) {
       return false;
     }
-    if (!info.modifiers.contains(HotKeyModifier.shift)) {
+    final modifierSet = info.modifiers.toSet();
+    if (modifierSet.length != 1 ||
+        !modifierSet.contains(HotKeyModifier.shift)) {
       return false;
     }
     return info.keyCode == arrowKey;

